### PR TITLE
gh-103406: Modernize pos-only arguments usage in `test_signature`

### PR DESCRIPTION
**Pull request by sobolevn**
We can be sure that these `.__signature__` manipulations are safe to removed, because:
1. Tests do pass now
2. Places where `.replace` calls are required by the test logic (for example, when testing `Signature` and `Parameter` subclasses or invalid states) are not touched
3. `.__signature__` change is tested here: https://github.com/python/cpython/blob/ecad802e3f9ef7063e7d7fe9f0c037561396ef8e/Lib/test/test_inspect.py#L1045-L1051

Now, `/` is used where it must be used.

* Issue: gh-103406

### CODE REVIEW

**Comment by sobolevn** [comment]
I am not sure about 3.10 backport 🤔

File: Lib/test/test_inspect.py [hunk 1]
```diff
@@ -3044,12 +3044,10 @@ def foo(a=1, b=2, c=3):
...
 
 
-        def foo(a, b, c, d, **kwargs):
+        def foo(a, b, /, c, d, **kwargs):
             pass
         sig = inspect.signature(foo)
         params = sig.parameters.copy()
-        params['a'] = params['a'].replace(kind=Parameter.POSITIONAL_ONLY)
-        params['b'] = params['b'].replace(kind=Parameter.POSITIONAL_ONLY)
         foo.__signature__ = inspect.Signature(params.values())
```
Comment (JelleZijlstra) [inline Lib/test/test_inspect.py]: I think these lines can also be removed.

File: Lib/test/test_inspect.py [hunk 2]
```diff
@@ -4157,16 +4150,9 @@ def test(a, *args, b, z=100, **kwargs):
     def test_signature_bind_positional_only(self):
         P = inspect.Parameter
 
-        def test(a_po, b_po, c_po=3, foo=42, *, bar=50, **kwargs):
+        def test(a_po, b_po, c_po=3, /, foo=42, *, bar=50, **kwargs):
             return a_po, b_po, c_po, foo, bar, kwargs
 
-        sig = inspect.signature(test)
```
Comment (JelleZijlstra) [inline Lib/test/test_inspect.py:4163]: I think it's worth adding a new test that's explicitly designed to check that if you modify `__signature__`, `inspect.signature` retrieves the modified signature. I don't think there's other tests in `test_inspect` that directly test that behavior.

Comment (sobolevn) [reply Lib/test/test_inspect.py:4163]: I think that this test already does this: https://github.com/python/cpython/blob/ecad802e3f9ef7063e7d7fe9f0c037561396ef8e/Lib/test/test_inspect.py#L1045-L1051 :)

I've mentioned it in my PR description 👍

**Comment by sobolevn** [comment]
Thank you for the review! 🎉

File: Lib/test/test_inspect.py (hunk 2 above)
Comment (JelleZijlstra) [reply Lib/test/test_inspect.py:4163]: I saw that but I feel it's testing `getfullargspec`, not `signature`.

Comment (sobolevn) [reply Lib/test/test_inspect.py:4163]: Ok, new tests are always a good thing, adding :)

Comment (sobolevn) [reply Lib/test/test_inspect.py:4163]: Done!

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**